app/models/manageiq/providers/base_manager/refresher.rb

```diff
@@ -239,52 +242,39 @@ def group_targets_by_ems(targets)
 
       # We preprocess targets to merge all non ExtManagementSystem class targets into one
       # InventoryRefresh::TargetCollection. This way we can do targeted refresh of all queued targets in 1 refresh
-      def preprocess_targets_manager_refresh
-        @targets_by_ems_id.each do |ems_id, targets|
-          ems = @ems_by_ems_id[ems_id]
-
-          # We want all targets of class EmsEvent to be merged into one target, so they can be refreshed together, otherwise
-          # we could be missing some crosslinks in the refreshed data
-          all_targets, sub_ems_targets = targets.partition { |x| x.kind_of?(ExtManagementSystem) }
-
-          if sub_ems_targets.present?
-            if ems.allow_targeted_refresh?
-              # We can disable targeted refresh with a setting, then we will just do full ems refresh on any event
-              ems_event_collection = InventoryRefresh::TargetCollection.new(:targets    => sub_ems_targets,
-                                                                            :manager_id => ems_id,
-                                                                            :manager    => ems)
-              all_targets << ems_event_collection
-            else
-              all_targets << @ems_by_ems_id[ems_id]
-            end
-          end
```

Comment:
This is mainly the reason why I wanted to refactor this, we end up checking if we should do a full refresh in 3 different places with nested conditionals